src/components/SessionCard/SessionCard.tsx

```diff
@@ -16,80 +17,92 @@ const weekday = [
   "Saturday",
 ];
 
-function checkMark(status: SessionCardStatus) {
+function CheckMark({ status }: { status: SessionCardStatus }) {
   return (
-    <>
-      {status == SessionCardStatus.SELECTED ? (
-        <div className="absolute right-20">
-          <svg
-            xmlns="http://www.w3.org/2000/svg"
-            viewBox="0 0 20 20"
+    <div className="absolute right-7 top-1/2 -translate-y-1/2">
+      {status === SessionCardStatus.SELECTED && (
+        <svg
+          xmlns="http://www.w3.org/2000/svg"
+          width="23"
+          height="23"
+          viewBox="0 0 23 23"
+          fill="none"
+        >
+          <path
+            d="M11.1591 0C5.01042 0 0 5.01042 0 11.1591C0 17.3077 5.01042 22.3181 11.1591 22.3181C17.3077 22.3181 22.3181 17.3077 22.3181 11.1591C22.3181 5.01042 17.3077 0 11.1591 0ZM16.4931 8.59247L10.1659 14.9197C10.0097 15.0759 9.79765 15.1652 9.57447 15.1652C9.35129 15.1652 9.13927 15.0759 8.98304 14.9197L5.82503 11.7616C5.50141 11.438 5.50141 10.9024 5.82503 10.5788C6.14864 10.2552 6.68427 10.2552 7.00789 10.5788L9.57447 13.1454L15.3102 7.40961C15.6338 7.086 16.1695 7.086 16.4931 7.40961C16.8167 7.73322 16.8167 8.2577 16.4931 8.59247Z"
             fill="white"
-            width="45"
-            height="45"
-          >
-            <path
-              fill-rule="evenodd"
-              d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.857-9.809a.75.75 0 00-1.214-.882l-3.483 4.79-1.88-1.88a.75.75 0 10-1.06 1.061l2.5 2.5a.75.75 0 001.137-.089l4-5.5z"
-              clip-rule="evenodd"
-            />
-          </svg>
-        </div>
-      ) : (
-        <></>
+          />
+        </svg>
       )}
-    </>
+    </div>
   );
 }
 
-const SessionCard = ({startdate, enddate, status, location}: SessionCardProps) => {
-  const dayOfWeek = weekday[startdate.getDay()];
-  const start_time = startdate
+const SessionCard = ({
+  startDate,
+  endDate,
+  status,
+  location,
+  onChange,
+}: SessionCardProps) => {
+  const dayOfWeek = weekday[startDate.getDay()];
+  const start_time = startDate
     .toLocaleTimeString([], { timeStyle: "short" })
     .toUpperCase();
-  const end_time = enddate
+  const end_time = endDate
     .toLocaleTimeString([], { timeStyle: "short" })
     .toUpperCase();
 
-  let cardClassName = "relative px-8 py-5 ";
-  let dayOfWeekClassName = "font-medium text-2xl ";
-  let locationClassName = "font-normal text-lg ";
-  let timeClassName = "font-normal text-lg pt-2 ";
+  let cardClassName;;
+  let dayOfWeekClassName;
+  let locationClassName;
+  let timeClassName;
 
   // Exact colours to be adjusted
   switch (status) {
```

Comment:
Oh my... I'm gonna go and refactor this myself. I'm having anxiety problems reading this